### ISSUE TITLE
Calamares doesn't create a proper BTRFS installation (no subvolumes)

### ISSUE THREAD

**Issue by DeskworkTrickster** [NixOS/calamares-nixos-extensions#12]
### Describe the bug
When using the Calamares GUI installer it doesn't seem to be possible to create a proper BTRFS installation with subvolumes.
Creating the partitions works fine. But there is no option to configure subvolumes and there are none created automatically.

### Steps To Reproduce
1. Download a 22.05 Live ISO
2. Use "manual partitioning" in the GUI installer to setup a BTRFS filesystem
3. Install the system

### Expected behavior
The system gets installed with BTRFS and subvolumes.

### Additional context
Consider making BTRFS (instead of EXT4) the default installation option when going for "erase disk", since it seems to become a wildly popular option for linux distros in general.

**Comment by etrigan63** [comment]
Yes, please! BTRFS offers a lot of features that would improve many aspects of NixOS! Compression per subvolume, snapshots, and many others. Be sure to also allow multiple drives to be incorporated. I have three drives in my main system (all BTRFS - one for /home and another for data /mnt/data besides the main drive at /).

**Comment by AldebaraanMKII** [comment]
Seconding this. BTRFS offers much more advantages over ext4 in this day and age of super fast cpus.

**Comment by ElvishJerricco** [comment]
This repo has been merged upstream into nixpkgs and will be archived. If this is still an issue, please open a new one there.
